src/credentials.cpp

```diff
@@ -194,7 +212,7 @@ namespace preferences
 	}
 }
 
-// TODO: No idea if this is a reasonable way of generating the key.
+// TODO: Key-stretching (would hashing the key with SHA512 or something work? bcrypt was recommended though)
```

Comment:
SHA512 isn't a good key stretching algorithm.

The point of key stretching is to slow down brute-forcing attempts. Because an attacker needs to stretch every key before trying it, trying all possible keys becomes infeasible if the number of possible keys is high enough.

SHA512 is designed as a fast hashing algorithm. It's way too fast to be useful for key stretching.